pgmq-extension/sql/pgmq--1.3.3--1.3.4.sql

```diff
@@ -0,0 +1,39 @@
+-- Add function to convert archive queues to partitioned tables
+CREATE pgmq.convert_archive_partitioned(table_name TEXT)
+RETURNS void AS $$
+BEGIN
+    EXECUTE format('ALTER TABLE pgmq.a_%I RENAME TO a_%I_old', table_name, table_name);
+
+    EXECUTE format('
+    CREATE TABLE pgmq.a_%I (
+        msg_id bigint NOT NULL,
+        read_ct int4 NULL DEFAULT 0,
+        enqueued_at timestamptz NULL DEFAULT now(),
+        archived_at timestamptz NULL DEFAULT now(),
+        vt timestamptz NULL,
+        message jsonb NULL
+    )
+    PARTITION BY RANGE (archived_at)', table_name);
+
+    EXECUTE format('ALTER INDEX pgmq.archived_at_idx_a_%I RENAME TO archived_at_idx_a_%I_old',
+                   replace(table_name, 'a_', ''), replace(table_name, 'a_', ''));
+    EXECUTE format('CREATE INDEX pgmqarchived_at_idx_a_%I ON pgmq.a_%I (archived_at)',
```

Comment:
I'm worried about max identifier name size and truncation here.

Reply:
I think here we can just error/fail if table name is too long by providing the same name check as we do with the queue name, via `PERFORM pgmq.validate_queue_name(queue_name)`?

Reply:
The thing is `validate_queue_name` assumes a max queue name of 48 or something (I don't recall the exact value), which is sufficient for the biggest identifier we create with a queue name, which, if I recall correctly, is something like `a_{queue_name}_archived_at_idx`. But if you append `_old` to it, you get 4 extra characters, which then may get truncated if the queue_name is >=45

Reply:
> The thing is `validate_queue_name` assumes a max queue name of ~48 (I don't recall the exact value), which is sufficient for the biggest identifier we create with a queue name, which, if I recall correctly, is something like `a_{queue_name}_archived_at_idx`. But if you append `_old` to it, you get 4 extra characters, which then may get truncated if the queue_name is >=45

Index pattern is 'colname_idx_{queue_name}' we don't append 'a_'  before queue_name for index name. I was wondering if we can delete the index on _old table?